spot_driver/launch/spot_driver.launch.py

```diff
@@ -104,9 +105,43 @@ def create_point_cloud_nodelets(
     return composable_node_descriptions
 
 
-def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
+def spot_has_arm(context: LaunchContext) -> bool:
+    """Check if spot has an arm by logging in and instantiating a SpotWrapper"""
+    config_file_path = LaunchConfiguration("config_file").perform(context)
+    spot_name = LaunchConfiguration("spot_name").perform(context)
     logger = logging.getLogger("spot_driver_launch")
+    username = os.getenv("BOSDYN_CLIENT_USERNAME")
+    password = os.getenv("BOSDYN_CLIENT_PASSWORD")
+    hostname = os.getenv("SPOT_IP")
+    port = int(os.getenv("SPOT_PORT", "0"))  # TODO should the user be able to specify a port via config file?
+    # parse the yaml to determine if login information is set there
+    if os.path.isfile(config_file_path):
+        with open(config_file_path, "r") as config_yaml:
+            try:
+                config_dict = yaml.safe_load(config_yaml)
+                if ("/**" in config_dict) and ("ros__parameters" in config_dict["/**"]):
+                    ros_params = config_dict["/**"]["ros__parameters"]
+                    # only set username/password/hostname if they were not already set as environment variables.
+                    if (not username) and ("username" in ros_params):
+                        username = ros_params["username"]
+                    if (not password) and ("password" in ros_params):
+                        password = ros_params["password"]
+                    if (not hostname) and ("hostname" in ros_params):
+                        hostname = ros_params["hostname"]
+            except yaml.YAMLError as exc:
+                print("Parsing config_file yaml failed with: {}".format(exc))
+    if (username is None) or (password is None) or (hostname is None):
+        raise ValueError(
+            "Login to Spot failed using [Username: '{}' Password: '{}' Hostname: '{}']. Update your config_file yaml"
+            " or ensure that your environment variables are set.".format(username, password, hostname)
+        )
```

Comment:
I think it would be nice to have this logic to load and parse the config file as a few separate function. It'll make `spot_has_arm` easier to read.

Maybe something like this (warning: pseudocode!):

```
def spot_has_arm:
  username, password, hostname, port = get_authentication_parameters()  # try parsing env vars and config file values in here
  spot_wrapper = SpotWrapper(
        username=username, password=password, hostname=hostname, port=port, robot_name=spot_name, logger=logger
  )
  return spot_wrapper.has_arm()
```